contracts/test/GrantRegistry.test.ts

```diff
@@ -0,0 +1,222 @@
+// --- External imports ---
+import { artifacts, ethers, waffle } from 'hardhat';
+import { BigNumber, BigNumberish } from 'ethers';
+import { Artifact } from 'hardhat/types';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+import { expect } from 'chai';
+
+// --- Our imports ---
+import { GrantRegistry } from '../typechain/GrantRegistry';
+
+// --- Parse and define helpers ---
+const { deployContract } = waffle;
+const { isAddress } = ethers.utils;
+const randomAddress = () => ethers.Wallet.createRandom().address;
+const randomPtr = () => String(Math.random()); // doesn't need to be a valid URL for testing
+const BN = (x: BigNumberish) => BigNumber.from(x);
+
+// --- Type definitions ---
+// The output from ethers/typechain allows array or object access to grant data, so we must define types for
+// handling the Grant struct as done below
+type GrantObject = {
```

Comment:
We'll almost certainly need these type definitions in other tests, so might be worth extracting them to a file and importing them now establish the pattern. Eventually, we may even want to share these test types with the frontend. Worth considering.

Reply:
i'm happy to create a shared types directory at the root, either here in this PR or as a followup. Let me know how I Can help.

Reply:
Shared types directory at the root is a good idea. Created this issue and assigned it to you @wildmolasses: https://github.com/dcgtc/dgrants/issues/8

For this PR, we don't currently have other tests that need these types, and don't know whether #8 or new contract tests will be done first, so seems we should leave this as-is for now to avoid work that may be unnecessary

Reply:
Sounds like a good plan. Feel free to tackle this at your convenience @wildmolasses. I've updated the PR description to note that existing tests should also be updated to use the shared type dir when the change is made.